api/v3/Payment.php

```diff
@@ -98,21 +98,26 @@ function civicrm_api3_payment_delete(&$params) {
  * @return array
  *   Api result array
  */
-function civicrm_api3_payment_cancel(&$params) {
+function civicrm_api3_payment_cancel($params) {
   $eftParams = array(
     'entity_table' => 'civicrm_contribution',
     'financial_trxn_id' => $params['id'],
   );
   $entity = civicrm_api3('EntityFinancialTrxn', 'getsingle', $eftParams);
-  $contributionId = $entity['entity_id'];
-  $params['total_amount'] = $entity['amount'];
-  unset($params['id']);
 
-  $trxn = CRM_Contribute_BAO_Contribution::recordAdditionalPayment($contributionId, $params, 'refund', NULL, FALSE);
+  $paymentParams = [
+    'total_amount' => -$entity['amount'],
```

Comment:
@eileenmcnaughton I think we shouldn't negate here? Test failure complains about the same

Reply:
@monishdeb you are right - but I think we should be setting to negative 'up high' and handling that lower down so I have removed the expectation of a positive number in 

https://github.com/civicrm/civicrm-core/pull/13689/files#diff-8967f6e57d6209aa277ded6512413608R316

and fixed the one other place that calls that to also pass in a negative number

https://github.com/civicrm/civicrm-core/pull/13689/files#diff-7a5b0e2d131dabc49178460fc63328c0R3882

There is still a test fail on status which I'm making sense of